From e4d227f5bd7a1abc084109dd43fe3be949bdaf1e Mon Sep 17 00:00:00 2001
Message-Id: <e4d227f5bd7a1abc084109dd43fe3be949bdaf1e.1687727967.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1687727967.git.sharlatanus@gmail.com>
References: <cover.1687727967.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 21 Jun 2023 22:45:13 +0100
Subject: [PATCH 7/9] gnu: xplanet: Use G-expressions.

* gnu/packages/astronomy.scm (xplanet): Use G-expressions, improve
package style.
[inputs]: Remove labels.
---
 gnu/packages/astronomy.scm | 39 +++++++++++++++++++-------------------
 1 file changed, 20 insertions(+), 19 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 310a6340f5..b212f4614f 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2720,28 +2720,29 @@ (define-public xplanet
          "xplanet-1.3.1-libimage_gif.c.patch"
          "xplanet-1.3.1-xpUtil-Add2017LeapSecond.cpp.patch"))))
     (build-system gnu-build-system)
+    (arguments
+     (list
+      #:configure-flags
+      #~(list
+         ;; no nasa jpl cspice support
+         "--without-cspice"
+         ;; Give correct path for pnm.h header to configure script
+         (string-append
+          "CPPFLAGS=-I" #$(this-package-input "netpbm") "/include/netpbm"))))
     (native-inputs
      (list pkg-config))
     (inputs
-     `(("libx11" ,libx11)
-       ("libxscrnsaver" ,libxscrnsaver)
-       ("libice" ,libice)
-       ("freetype" ,freetype)
-       ("pango" ,pango)
-       ("giflib" ,giflib)
-       ("libjpeg" ,libjpeg-turbo)
-       ("libpng" ,libpng)
-       ("libtiff" ,libtiff)
-       ("netpbm" ,netpbm)
-       ("zlib" ,zlib)))
-    (arguments
-     `(#:configure-flags
-       (let ((netpbm (assoc-ref %build-inputs "netpbm")))
-         (append (list
-                  ;; Give correct path for pnm.h header to configure script
-                  (string-append "CPPFLAGS=-I" netpbm "/include/netpbm")
-                  ;; no nasa jpl cspice support
-                  "--without-cspice" )))))
+     (list libx11
+           freetype
+           giflib
+           libice
+           libjpeg-turbo
+           libpng
+           libtiff
+           libxscrnsaver
+           netpbm
+           pango
+           zlib))
     (home-page "https://xplanet.sourceforge.net/")
     (synopsis "Planetary body renderer")
     (description
-- 
2.40.1

